businessRules/donations.js

```diff
@@ -1,54 +1,67 @@
 const { v4 } = require('uuid');
 const { createDonation } = require('../models/donations');
-const { getAllIdeas } = require('../models/ideas');
-const { getIdeas } = require('../utils/storage');
+const { findIdea } = require('../validations/ideas');
+const {
+    validateSum,
+    findIdeaWithTotalDonationSum,
+} = require('../validations/donations.js');
+const { findUser } = require('../validations/users');
+const { getUsers } = require('../utils/storage');
 
-function createDonation(firstName, sum, userId, ideaId) {
-    let ideas = getIdeas();
-    const idea = ideas.find((idea) => ideaId === idea.id);
+function createDonationByUnregisteredUser(sum, userId, ideaId) {
+    findIdea(ideaId);
```

Comment:
I would expect `findIdea` function to return an idea, as the name suggests. It's a "getter" function, but you're not "getting" anything  back here.